packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.tsx

```diff
@@ -232,17 +233,30 @@ export function useGridColumns(
     (field, width) => {
       logger.debug(`Updating column ${field} width to ${width}`);
 
-      const column = apiRef.current.getColumn(field);
-      const newColumn = { ...column, width };
-      apiRef.current.updateColumns([newColumn]);
+      const columnsState = gridColumnsStateSelector(apiRef.current.state);
+      const column = columnsState.lookup[field];
+      const newColumn: GridStateColDef = { ...column, width, hasBeenResized: true };
+
+      setGridColumnsState(
+        hydrateColumnsWidth(
+          {
+            ...columnsState,
+            lookup: {
+              ...columnsState.lookup,
+              [field]: newColumn,
+            },
+          },
+          apiRef.current.getRootDimensions?.()?.viewportInnerSize.width ?? 0,
```

Comment:
Is the `?.` after `getRootDimensions` necessary?

Reply:
It's unnecessary indeed, thanks for spotting it 👍🏻